lms/services/canvas_api/client.py

```diff
@@ -200,23 +201,45 @@ def post_load(self, data, **_kwargs):  # pylint:disable=no-self-use
                 for enrollment in data["enrollments"]
             ]
 
-    def list_files(self, course_id):
+    def list_files(self, course_id, request=None):
         """
         Return the list of files for the given `course_id`.
 
         :param course_id: the Canvas course_id of the course to look in
+        :param request: Pyramid request object used to store file details if
+            present
         :rtype: list(dict)
         """
         # For documentation of this request see:
         # https://canvas.instructure.com/doc/api/files.html#method.files.api_index
 
-        return self._client.send(
+        files = self._client.send(
             "GET",
             f"courses/{course_id}/files",
             params={"content_types[]": "application/pdf"},
             schema=self._ListFilesSchema,
         )
 
+        if request:
```

Comment:
Should this just be mandatory? Seems like this is a nice way to try and keep Pyramid from being a mandatory part of using this.

Reply:
I don't have a better suggestion (a persist: bool parameter and the request seems worse) but seems odd that passing a `request` object implies persisting the object

Reply:
Can't the service factory inject `request.registry.notify` into the service?

Reply:
We can inject `request.registry.notify`, but we still need `request` around to put into the event. So I've injected the whole request instead.